### PR TITLE
FIX: load existing dm channel via url

### PR DESCRIPTION
**Pull request by dbattersby**
When users click a link that points to an existing group chat, we should reopen that chat instead of creating a new group chat so users can more easily continue ongoing conversations.

### CODE REVIEW

File: plugins/chat/app/services/chat/create_direct_message_channel.rb [hunk 1]
```diff
@@ -42,6 +42,7 @@ class Contract
       attribute :name, :string
       attribute :target_usernames, :array
       attribute :target_groups, :array
+      attribute :chat_link, :boolean, default: false
```
Comment (jjaffeux) [inline plugins/chat/app/services/chat/create_direct_message_channel.rb:45]: It's not very clear to me what chat_link is going to do just from reading the name

Comment (jjaffeux) [reply plugins/chat/app/services/chat/create_direct_message_channel.rb:45]: Maybe `upsert` true/false, would be a better option name?

File: plugins/chat/app/services/chat/create_direct_message_channel.rb [hunk 2]
```diff
@@ -42,6 +42,7 @@ class Contract
       attribute :name, :string
       attribute :target_usernames, :array
       attribute :target_groups, :array
+      attribute :upsert, :boolean, default: false
```
Comment (jjaffeux) [inline plugins/chat/app/services/chat/create_direct_message_channel.rb:45]: missing upsert documentation above

File: plugins/chat/spec/requests/chat/api/direct_messages_controller_spec.rb [hunk 3]
```diff
@@ -78,13 +78,25 @@ def create_dm_channel(user_ids)
...
-      it "createsa new dm channel" do
+      it "creates a new dm channel" do
         create_dm_channel(direct_message_user_ids)
 
         expect {
           post "/chat/api/direct-message-channels.json", params: { target_usernames: [usernames] }
         }.to change { Chat::DirectMessage.count }.by(1)
       end
+
+      it "returns existing dm channel when using chat link" do
```
Comment (jjaffeux) [inline plugins/chat/spec/requests/chat/api/direct_messages_controller_spec.rb:89]: I think it should be more ` it "returns existing dm channel when upsert is true" do`

**Review by jjaffeux — APPROVED** [review]

File: plugins/chat/spec/requests/chat/api/direct_messages_controller_spec.rb (hunk 3 above)
Comment (dbattersby) [reply plugins/chat/spec/requests/chat/api/direct_messages_controller_spec.rb]: Thanks, I missed that one 👍

**Review by jjaffeux — APPROVED** [review]

**Review by ZogStriP — COMMENTED** [review]
LGTM 👍

**Pull request merged**